REPOSITORY_STANDARDS.md

```diff
@@ -0,0 +1,129 @@
+
+## Contents
+
+- [Naming conventions](#naming-conventions)
+- [Visibility](#visibility)
+- [Repository documentation](#repository-documentation)
+- [Permissions](#permissions)
+
+## Naming conventions
+
+Repositories should be named with the prefix `dis-` to distinguish them as belonging to `Dissemination`. This helps to identify our repositories amongst the many that are part of ONSDigital.
+
+Historical repositories have the prefix `dp-` as they belonged to `Digital Publishing`.
+
+### Branch naming
+
+#### Protected branches
+
+`main`: This is the 'live' branch and is the single source of truth showing the release history for the repo.
+`develop`: This branch is present on repos that have opted to adopt a GitFlow branching strategy and is the integration branch where fixes and features are integrated prior to a release branch being cut
+
+`master` has been historically used in older repositories and should be migrated to `main` when feasible.
+
+#### Development branches
+
+`feature/*` - for features
+`fix/*` - for fixes
+`hotfix/*` - for hotfixes from `main`
+`release/*` - for cutting releases
+
+## Visibility
+
+We aim to work in the open where possible and so the majority of our repositories should be `public`.
+
+Repositories should be marked as `private` if they reveal details of:
+
+- ONS infrastructure
+- encrypted secrets files
+
+### GitHub Copilot
+
+If a repo is `private` then GitHub Copilot and similar tools should be excluded from reading the content in the repository.
+
+## Repository documentation
+
+### CODEOWNERS
+
+Every repository should have a CODEOWNERS file in it's `.github` folder. This should show the responsible team(s) for that repository.
+
+### CONTRIBUTING
+
+Every repository should have a CONTRIBUTING file in the root directory. This should make clear to engineers how they can contribute to that repository, including:
+
+- branching strategy
+- commit approach
+- merging strategy
+
+### LICENSE
+
+Every repository should have a LICENSE file in the root directory. As standard we use the MIT licencse.
+
+### PULL_REQUEST_TEMPLATE
+
+Every repository should have a PULL_REQUEST_TEMPLATE file in it's `.github` folder. This should make clear how pull requests are documented for that repository.
+
+### README
+
+Every repository should have a README file in the root directory.
+
+For applications, this should include:
+
+- how to run the service locally, if applicable
+- details of environmental configuration
+- dependencies required to run in order to develop locally
+- where appropriate, a link to the docker-compose stack to run it with other services
+
+### Makefile
+
+Every repository should use a Makefile for running:
+
+- tests
+- builds
+- linters
+- running application
+
+Where possible, these should be consistent across applications / libraries. Some common targets include:
+
+- build
+- debug
+- lint
+- test
+- test-component
```

Comment:
Brief descriptions for what each one of these is used for could help ensure consistency, e.g.:
```suggestion
- build: used to build the project ready for deployment
- debug: used to build and run the project locally often watching for local changes and rebuilding to aid in local development
- lint: runs all linters and other static code analysis tools for the project
- test: runs unit and integration tests for the project
- test-component: runs component tests for the project
```